pinot-query-runtime/src/main/java/org/apache/pinot/query/mailbox/InMemorySendingMailbox.java

```diff
@@ -87,9 +89,10 @@ private void sendPrivate(MseBlock block, List<DataBuffer> serializedStats)
       case CANCELLED:
         throw new QueryCancelledException(String.format("Mailbox: %s already cancelled from upstream", _id));
       case ERROR:
-        throw new RuntimeException(String.format("Mailbox: %s already errored out (received error block before)", _id));
+        throw new QueryException(QueryErrorCode.ERRORED_OUT, String.format(
+            "Mailbox: %s already errored out (received error block before)", _id));
```

Comment:
I would prefer not to introduce new error codes if unnecessary. Are we sure we need this one here? Couldn't we just report it as internal?

Reply:
Yeah, just replaced with `INTERNAL`.